File: go/implode.go
package pehape

import (
	"fmt"
)

//  The Implode function joins array/slice element with a string.
//  Parameter :
//  - array => The array/slice to join to a string.
//  - separator => Specifies what to put between the array elements.
//  Return :
//  - The string with all array/slice elements joined.
func Implode[T any](array []T, sep ...string) string {
	var result string
	for i := 0; i < len(array); i++ {
		if sep == nil || len(sep) == 0 {
			result = result + fmt.Sprint(array[i])
		} else {
			if i == 0 {
				result = fmt.Sprint(array[i])
			} else {
				result = result + sep[0] + fmt.Sprint(array[i])
			}
		}
	}
	return result
}

File: csharp/Pehape/String/Implode.cs
using System.Collections.Generic;
using System.Linq;

namespace Pehape {
	// ReSharper disable once InconsistentNaming
	public static partial class PHP {
		/// <summary>
		/// The Implode function joins array elements with a string.
		/// </summary>
		/// <typeparam name="T">The type of members of array.</typeparam>
		/// <param name="separator">Specifies what to put between the array elements.</param>
		/// <param name="array">The array to join to a string.</param>
		/// <returns>A string from elements of an array</returns>
		public static string Implode<T>(string? separator, IEnumerable<T> array) => string.Join(separator, array);

		/// <summary>
		/// The Implode function joins array elements with a string.
		/// </summary>
		/// <typeparam name="T">The type of members of array.</typeparam>
		/// <param name="separator">Specifies what to put between the array elements.</param>
		/// <param name="array">The array to join to a string.</param>
		/// <returns>A string from elements of an array</returns>
		public static string Implode<T>(string? separator, IEnumerable<T[]> array) =>
			string.Join(separator, array.SelectMany(x => x));

		/// <summary>
		/// The Implode function joins array of elements with a string.
		/// </summary>
		/// <typeparam name="T">The type of members of array.</typeparam>
		/// <param name="separator">Specifies what to put between the array elements.</param>
		/// <param name="array">The array to join to a string.</param>
		/// <returns>A string from elements of an array</returns>
		public static string Implode<T>(char separator, IEnumerable<T> array) => string.Join(separator, array);

		/// <summary>
		/// The Implode function joins array of elements with a string.
		/// </summary>
		/// <typeparam name="T">The type of members of array.</typeparam>
		/// <param name="separator">Specifies what to put between the array elements.</param>
		/// <param name="array">The array to join to a string.</param>
		/// <returns>A string from elements of an array</returns>
		public static string Implode<T>(char separator, IEnumerable<T[]> array) =>
			string.Join(separator, array.SelectMany(x => x));
	}
}
